src/QsCompiler/DocumentationGenerator/Extensions.cs

```diff
@@ -0,0 +1,419 @@
+        /// </summary>
+        /// <typeparam name="T">The type of AST item decorated by the attribute builder.</typeparam>
+        /// <param name="builder">The attribute builder to which the new attributes should be added.</param>
+        /// <param name="attributeName">The name of the Q# UDT for the new attributes.</param>
+        /// <param name="items">The values of the new attributes to be added.</param>
+        /// <returns>A new attribute builder with the one new attribute added for each element of <paramref name="items"/>.</returns>
+        internal static IAttributeBuilder<T> WithListOfDocumentationAttributes<T>(
+            this IAttributeBuilder<T> builder, string attributeName, IEnumerable<string> items
+        ) =>
+            items
+            .Aggregate(
+                builder,
+                (acc, item) => acc.WithDocumentationAttribute(
+                    attributeName, item.AsLiteralExpression()
+                )
+            );
+
+        internal static IAttributeBuilder<T> WithDocumentationAttributesFromDictionary<T>(
+            this IAttributeBuilder<T> builder, string attributeName, IDictionary<string, string> items
+        ) =>
+            items
+            .Aggregate(
+                builder,
+                (acc, item) => acc.WithDocumentationAttribute(
+                    attributeName,
+                    // The following populates all of the metadata needed for a
+                    // Q# literal of type (String, String).
+                    new TypedExpression(
+                        QsExpressionKind<TypedExpression, Identifier, ResolvedType>.NewValueTuple(
+                            ImmutableArray.Create(
+                                item.Key.AsLiteralExpression(),
+                                item.Value.AsLiteralExpression()
+                            )
+                        ),
+                        ImmutableArray<Tuple<QsQualifiedName, NonNullable<string>, ResolvedType>>.Empty,
+                        ResolvedType.New(
+                            QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>.NewTupleType(
+                                ImmutableArray.Create(
+                                    ResolvedType.New(QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>.String),
+                                    ResolvedType.New(QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>.String)
+                                )
+                            )
+                        ),
+                        new InferredExpressionInformation(false, false),
+                        QsNullable<Range>.Null
+                    )
+                )
+            );
+
+        internal static string ToSyntax(this ResolvedType type) =>
+            SyntaxTreeToQsharp.Default.ToCode(type);
+
+        internal static string ToSyntax(this QsTypeItem item) =>
+            item switch
```

Comment:
A bunch of your switch expressions don't cover all cases (the default usually is missing). Please add that; ideally you could even generate a proper diagnostic for the rewrite step in that case.